packages/react/src/errorboundary.tsx

```diff
@@ -66,7 +66,7 @@ const INITIAL_STATE = {
  * Sentry React SDK ErrorBoundary caught an error invoking your application code. This
  * is expected behavior and NOT indicative of a bug with the Sentry React SDK.
  */
-class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+class ErrorBoundary extends React.Component<React.PropsWithChildren<ErrorBoundaryProps>, ErrorBoundaryState> {
```

Comment:
Hi @cameronaziz, thank you so much for submitting this PR. We would like to fix this type error and release a patch.

Would you be willing to get rid of the `React.PropsWithChildren` helper here and instead add a `children?: React.ReactNode` prop to `ErrorBoundaryProps`, just like one of the maintainers of react suggests [here](https://stackoverflow.com/questions/71788254/react-18-typescript-children-fc/71809927#71809927)? If that's done we're good to go!

Reply:
@lforst Are you thinking more on the direction of what my original commit was: https://github.com/getsentry/sentry-javascript/pull/4966/commits/ea99673afb30bf5f77d842d0ab211c0ff7a4f912

The issue was it would not build with the children render as a function.

Reply:
1a575e68fd58d64f0dfde90ea7990c48ee07db2b without the `| undefined` would be perfect imo.

That should be fine typewise. Seems weird that it's not building. Can you update the code? I'll take a look at the build output afterwards.

Reply:
I can, but that ends up making the child within this block, https://github.com/getsentry/sentry-javascript/pull/4966/files#diff-0c3aa317cf9c841d89853ace58e8572990bb3c464066708a4c08242afe9e556bL154, a `never`

Reply:
You're right. Apparently `@types/react@18` changed the `ReactNode` type not to be callable anymore. I think up to v17 it had a constructor type of some sort. For me, when using the types from `@type/react@18` neither `children?: ReactNode` nor `React.PropsWithChildren` works. I think it's best if we go with the changes from ea99673afb30bf5f77d842d0ab211c0ff7a4f912.

Sorry for flip-flopping around. Thank you very much for helping out 🙏